pandera/engines/pandas_engine.py

```diff
@@ -737,6 +737,12 @@ def _coerce(
 
         def _to_datetime(col: PandasObject) -> PandasObject:
             col = to_datetime_fn(col, **self.to_datetime_kwargs)
+            if (
+                hasattr(pandas_dtype, "tz")
+                and pandas_dtype.tz is not None
+                and col.dt.tz is None
+            ):
+                col = col.dt.tz_localize(pandas_dtype.tz)
```

Comment:
@cristianmatache can we add a test case that executes this conditional? i.e. a case where the schema dtype is timezone aware but the data column isn't?